numpy/lib/tests/test_function_base.py

```diff
@@ -1510,6 +1510,17 @@ class A(np.ndarray):
         assert_(not isinstance(digitize(b, a, False), A))
         assert_(not isinstance(digitize(b, a, True), A))
 
+    def test_large_integers_increasing(self):
+        # gh-11022
+        x = 2**54  # loses precision in a float
+        assert_equal(np.digitize(x, [x - 1, x + 1]), 1)
+
+    @dec.knownfailureif(True, "np.core.multiarray._monoticity loses precision")
```

Comment:
This needs:
```python
from numpy.testing import dec
```

I took a look at the source for `knownfailureif`, and it is using `nose` under the hood. I'd normally suggest using `@pytest.mark.xfail` instead, but NumPy roadmap seems to imply that we want to avoid using "pytest magic" and mostly just use it as a runner (it seems [Guido agrees](https://twitter.com/gvanrossum/status/743107058778546177), but if nose is unmaintained may need some thought). I'm not sure if that nose stuff behind `dec` (and perhaps elsewhere?) will eventually have to be replaced though.

Reply:
> NumPy roadmap seems to imply that we want to avoid using "pytest magic" and mostly just use it as a runner

Don't use anything with nose in it, we don't want the dependency, nose itself is unmaintained, and the the version up on pip is not python 3.7 compatible. Definitely use `xfail` instead.

Reply:
We keep `dec` for backwards compatibility with folks using the numpy testing framework, you will note that numpy itself no longer uses it anywhere except for testing testing.

Reply:
Guido has a point about the pytest "documentation by example", the voluminous documentation for pytest is very hard to use for reference and learning, but I expect that at some point someone will make a "real" reference :)

Reply:
unless there's an equally concise alternative, I don't see a problem with `@pytest.mark.xfail`. The `xfail` and `skip` marks are kind of essential.